adagrams/game.py

```diff
@@ -1,11 +1,159 @@
+        letter_drawn = list_letter_drawn[0]
+        if letter_drawn in letter_freq:
+            if letter_freq[letter_drawn] < LETTER_POOL[letter_drawn]:
+                letter_freq[letter_drawn] +=1
+            else:
+                continue
+        else:
+            letter_freq[letter_drawn] = 1
+
+        ten_letters.append(letter_drawn)
+
+
+    return ten_letters        
+
 
 def uses_available_letters(word, letter_bank):
-    pass
+#  """Has two parameters:
+#    - `word`, the first parameter, describes some input word, and is a string
+#    - `letter_bank`, the second parameter, describes an array of drawn letters in a hand. You can expect this to be an array of ten strings, with each string representing a letter
+#     - Returns either `True` or `False`
+#     - Returns `True` if every letter in the `input` word is available (in the right quantities) in the `letter_bank`
+#     - Returns `False` if not; if there is a letter in `input` that is not present in the `letter_bank` or has too much of compared to the `letter_bank`
+# """ 
+    letter_bank_count = {}
+
+    for letters in letter_bank:
+        if letters in letter_bank_count:
+            letter_bank_count[letters] +=1
+        else:
+            letter_bank_count[letters] = 1
+
+    word_capital = word.upper()
+    for letter in word_capital:
+        if letter in letter_bank and letter_bank_count[letter]>=1:
+            letter_bank_count[letter] -=1  
+        else:
+            return False
+    return True
 
 def score_word(word):
-    pass
+# - Has one parameter: `word`, which is a string of characters
+# - Returns an integer representing the number of points
+# - Each letter within `word` has a point value. The number of points of each letter is summed up to represent the total score of `word`
+# - Each letter's point value is described in the table below
+# - If the length of the word is 7, 8, 9, or 10, then the word gets an additional 8 points
+# |Letter                        | Value|
+# |:----------------------------:|:----:|
+# |A, E, I, O, U, L, N, R, S, T  |   1  |
+# |D, G                          |   2  |
+# |B, C, M, P                    |   3  |
+# |F, H, V, W, Y                 |   4  |
+# |K                             |   5  |
+# |J, X                          |   8  |
+# |Q, Z                          |   10 |
+ 
+    point_system ={
+    "A": 1, 
+    "E": 1,
+    "I": 1, 
+    "O": 1, 
+    "U": 1,
+    "L": 1,
+    "N": 1,
+    "R": 1,
+    "S": 1,
+    "T": 1,
+    "D": 2, 
+    "G": 2,
+    "B": 3, 
+    "C": 3, 
+    "M": 3, 
+    "P": 3,
+    "F": 4, 
+    "H": 4, 
+    "V": 4, 
+    "W": 4, 
+    "Y": 4,
+    "K": 5,
+    "J": 8, 
+    "X": 8,
+    "Q": 10,
+    "Z": 10
+    }
+# FUNCTION BODY
+    word_capital = word.upper()
+    points = 0
+    for letter in word_capital:
+        points += point_system.get(letter)
+    if len(word_capital) in [7,8,9,10]:
+        points += 8
+    return points
```

Comment:
Excellent choice with this approach. I noticed you did a check for if the length of the word in between 7 and 10. We can also combine conditionals in one by checking if the length of the word is greater than 6 as well as less than 11 🎉